tools/internal/parser/validate.go

```diff
@@ -37,3 +45,137 @@ func (p *parser) requirePrivateDomainEmailContact() {
 		}
 	}
 }
+
+const (
+	amazonSuperblockStart = "Amazon : https://www.amazon.com"
+	amazonSuperblockEnd   = "concludes Amazon"
+)
+
+// requireSortedPrivateSection verifies that the blocks in the private
+// domains section is sorted according to PSL policy.
+func (p *parser) requireSortedPrivateSection() {
+	type chunk struct {
+		Name     string
+		Suffixes []Suffixes
+	}
+
+	coll := newCollator()
```

Comment:
Sorry, I don't really know Go. How do I figure out what this does?

Reply:
The code wasn't polished for review yet, sorry. I moved all the collation stuff to unicode.go and added a comment that gives a quick intro to sorting Unicode data and explains why we configure collation the way we do.